DM-818 Documentation for Loriot Downlink Support (#616)
SoftwareAG/c8y-docs: content/protocol-integration/lora-loriot-bundle/configure-loriot-credentials-cumulocity.md

---
weight: 25

> title: Configure the Loriot agent endpoint credentials and register the devices via Cumulocity

Reviewer: this title is three lines in the navigation. can it be shortened?

Author: Will try to shorten it

Author: Shortened now

layout: redirect
---

> ## <a name="cumulocity-loriot-connection-configuration">Creation of Loriot LNS Connection in Cumulocity</a>

Reviewer: ```suggestion
<a name="cumulocity-loriot-connection-configuration">
### Creation of Loriot LNS Connection in Cumulocity</a>
```

Before using LoRa devices with {{< product-c8y-iot >}}, you must configure the {{< product-c8y-iot >}} Loriot agent endpoint details in the Administration application. Click the **Connectivity** tab in the **Settings** menu to create, edit, delete or update multiple Loriot connections.

<a name="add-new-connection"></a>

> ### To add a new connection

Reviewer: ```suggestion
#### To add a new connection
```

> If you select **Connectivity** for the first time, you are asked to create a connection. Click **Add Connection**.

Reviewer: ```suggestion
When you select **Connectivity** for the first time, you are asked to create a connection. Click **Add Connection**.
```

Enter the following information:

- **Name**: The name of the Loriot connection being created.

> - **Base URL**: The URL associated with the Loriot Provider account.

Reviewer: ```suggestion
- **Base URL**: The URL associated with the Loriot provider account.
```

- **Username**: Your Loriot account username.
- **Password**: Your Loriot account password.

> Click **Save**. If you have entered the correct information, you see the message "Connection created".

Reviewer: ```suggestion
Click **Save**. If the information you have entered is correct, the message "Connection created" appears.
```

To add another connection, click **Add Connection** and follow the steps above.

> Note: Always keep the **Gateway Information** option enabled because the Loriot agent only processes "gw" (gateway information) messages.

Reviewer: ```suggestion
{{< c8y-admon-info >}}
Always keep the **Gateway Information** option enabled because the Loriot agent only processes "gw" (gateway information) messages.
{{< /c8y-admon-info >}}
```

![Enable gateway information option](/images/device-protocols/lora-loriot/loriot-gateway-option-enabled.png)

<a name="update-credentials-in-a-connection"></a>

> ### To update a connection

Reviewer: ```suggestion
#### To update a connection
```

Select the connection to be updated, make your edits, and save the connection.

> If there are devices associated with the connection, an error message will appear, stating "Can not update the LNS Connection with `<name of LNS Connection>` as it's associated with `<number of devices>`. Click the link to download the file with the details of the associated devices: `/service/<agent-context-path>/lns-connection/<lns-connection-name>/device`".

Reviewer: ```suggestion
If there are devices associated with the connection, an error message appears, stating "Can not update the LNS Connection with `<name of LNS Connection>` as it's associated with `<number of devices>`. Click the link to download the file with the details of the associated devices: `/service/<agent-context-path>/lns-connection/<lns-connection-name>/device`".
```
there's a mistake in the error message, but i guess we cannot fix it here? (Can not -> Cannot)

![Update connection information](/images/device-protocols/lora-loriot/loriot-admin-settings-update.png)

<a name="delete-connection"></a>

> ### To delete a connection

Reviewer: ```suggestion
#### To delete a connection
```

Select the connection to be deleted and click **Delete**.

> If there are devices associated with the connection, an error message will appear, stating "Can not delete the LNS Connection with `<name of LNS Connection>` as it's associated with `<number of devices>`. Click the link to download the file with the details of the associated devices: `/service/<agent-context-path>/lns-connection/<lns-connection-name>/device`".

Reviewer: ```suggestion
If there are devices associated with the connection, an error message appears, stating "Can not delete the LNS Connection with `<name of LNS Connection>` as it's associated with `<number of devices>`. Click the link to download the file with the details of the associated devices: `/service/<agent-context-path>/lns-connection/<lns-connection-name>/device`".
```

![Delete connection](/images/device-protocols/lora-loriot/loriot-admin-settings-delete.png)

> ## <a name="loriot-device-registration">Loriot device registration via API</a>

Reviewer: ```suggestion
<a name="loriot-device-registration">
### Loriot device registration via API</a>
```

After the Loriot LNS connection is created, the device can be registered using the REST API, whose endpoint and POST body are as follows:

> POST {{url}}/service/loriot/newDeviceRequest

Reviewer: ```suggestion
POST <kbd>{{url}}/service/loriot/newDeviceRequest</kbd>
```

> The body of the POST Request is as shown below:

Reviewer: ```suggestion
The body of the POST request:
```

---
weight: 25

> title: Device registration via Cumulocity

Reviewer: ```suggestion
title: Device registration via Cumulocity IoT
```
